### PR TITLE
Replace viz getters with native getters

### PR DESCRIPTION
**Pull request by davidmanzanares**
- Removed viz getFilter... getters
- Support setting viz properties directly: `viz.width = 7`

Covers: https://github.com/CartoDB/carto-vl/issues/406

### CODE REVIEW

**Comment by davidmanzanares** [comment]
I think it's OK, but it is quite a dangerous change, pls, review extra carefully.

File: src/api/viz.js [hunk 1]
```diff
@@ -53,38 +53,76 @@ export default class Viz {
...
     }
-
-    get resolution(){
-        return this._resolution;
-    }
-    set resolution(x){
-        this._resolution = x;
-        this._changed();
+    _defineProperty(propertyName, propertyValue) {
+        if (SUPPORTED_PROPERTIES.includes(propertyName)) {
```
Comment (IagoLast) [inline src/api/viz.js]: [replace nested conditional with guard conditions](https://refactoring.com/catalog/replaceNestedConditionalWithGuardClauses.html)

File: src/api/viz.js [hunk 2]
```diff
@@ -53,38 +53,76 @@ export default class Viz {
...
         this._validateAliasDAG();
     }
-
-    get resolution(){
-        return this._resolution;
-    }
-    set resolution(x){
-        this._resolution = x;
-        this._changed();
+    _defineProperty(propertyName, propertyValue) {
```
Comment (IagoLast) [inline src/api/viz.js]: A summary of this method would be nice

File: src/api/viz.js [hunk 3]
```diff
@@ -53,38 +53,76 @@ export default class Viz {
...
+        if (SUPPORTED_PROPERTIES.includes(propertyName)) {
+            Object.defineProperty(this, propertyName, {
+                get: () => this['__' + propertyName],
```
Comment (IagoLast) [inline src/api/viz.js]: Any special reason for the 2 underscores instead a single one?

File: test/integration/render/scenarios/variables/setter/scenario.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+// Check that precision is good
+const viz = new carto.Viz(`
+    @oneHundred: 10
+    width: ramp(linear($numeric, 0, 10), [0.10,0.20,0.30]) * @oneHundred
+`);
+const layer = new carto.Layer('layer', source, viz);
+
+layer.addTo(map);
+
+viz.variables.oneHundred = 100;
```
Comment (IagoLast) [inline test/integration/render/scenarios/variables/setter/scenario.js]: missing newline, you can configure this with [editorconfig](http://editorconfig.org/)

**Pull request merged**